app/controllers/tags_controller.rb

```diff
@@ -3,6 +3,7 @@ def from_dream
     dream = Dream.find(params['dream_id'])
     themes = dream.themes
     @tags = themes.map { | theme | theme.tag }
```

Comment:
it looks like you have the line `has_many :tags, through: :themes` in your `Dream` model.

this means we can replace lines 4 and 5 with: `@tags = dream.tags`. this'll be a much more efficient query than iterating through each theme to find it's tag